src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java

```diff
@@ -434,14 +393,52 @@ private static void skipArrayValue(MaskingState maskingState) {
     }
 
     /**
-     * Checks if the byte at the current index in the {@link MaskingState} is an unescaped double quote character in
-     * UTF-8.
+     * Expands the index from a double quote (current index) until end of the non-escaped double quote (including the
+     * double quote).
      *
      * @param maskingState the current {@link MaskingState}
-     * @return whether the byte at the current index is an unescaped double quote
+     * @return the amount of non-visible characters in the string - escape characters, utf-8 character data ('\u0000'),
+     * characters that use more than a single byte
      */
-    private static boolean currentByteIsUnescapedDoubleQuote(MaskingState maskingState) {
-        return AsciiCharacter.isDoubleQuote(maskingState.byteAtCurrentIndex())
-                && !AsciiCharacter.isEscapeCharacter(maskingState.byteAtCurrentIndexMinusOne());
+    private static int expandStringValue(MaskingState maskingState) {
+        maskingState.incrementCurrentIndex(); // step over the JSON key opening quote
+        int nonVisibleCharacters = 0;
+        boolean isEscapeCharacter = false;
+        boolean previousCharacterCountedAsEscapeCharacter = false;
+        while (!AsciiCharacter.isDoubleQuote(maskingState.byteAtCurrentIndex())
+               || (AsciiCharacter.isDoubleQuote(maskingState.byteAtCurrentIndex()) && isEscapeCharacter)) {
+            if (Utf8Util.getCodePointByteLength(maskingState.byteAtCurrentIndex()) > 1) {
```

Comment:
Ideally we should not count the non-visible characters for every string expansion, it only makes sense when we mask string characters, not the whole string. Not sure how much is the performance impact

Reply:
Did that in the latest commit, now we skip string without counting the non-visible characters and only count if when we have specific configuration

Reply:
Makes sense!